draft-annevk-johannhof-httpbis-cookies.md

```diff
@@ -1169,21 +1200,24 @@ boolean _httpOnlyAllowed_, boolean _allowNonHostOnlyCookieForPublicSuffix_, and
 
     1. _cookie_'s path's size is 1 and _cookie_'s path[0] is the empty string,
 
-   then return.
+   then return null.
 
 1. If _cookie_'s name is the empty byte sequence and one of the following is true:
 
     * _cookie_'s value, byte-lowercased, starts with `__secure-`, or
 
     * _cookie_'s value, byte-lowercased, starts with `__host-`
 
-   then return.
+   then return null.
 
 1. If the user agent's cookie store contains a cookie _oldCookie_ whose name is _cookie_'s name,
    host is host-equal to _cookie_'s host, host-only is _cookie_'s host-only, and path is path-equal to _cookie_'s path:
 
     1. If _httpOnlyAllowed_ is false and _oldCookie_'s http-only is true,
-       then return.
+       then return null.
+
+    1. If _cookie_'s secure flag is equal to _oldCookie_'s secure flag, _cookie_'s same-site is equal to _oldCookie_'s
+       same-site, and _cookie_'s expiry-time is equal to _oldCookie_'s expiry-time, then return null.
```

Comment:
We should probably write a test that if the expiry-time differs but everything else is the same you do get the event. Unless this is tested already?

Reply:
I could not find anything in https://github.com/web-platform-tests/wpt/tree/master/cookie-store, I think it is worth adding something for that and also testing that setting an already-expired cookie only triggers a change if it overwrites an existing one.